mate-scripts/src/services/Report.service.ts

```diff
@@ -0,0 +1,161 @@
+import path from 'path';
+import fs from 'fs-extra';
+import open from 'open';
+import { execBashCodeAsync } from '../tools';
+
+export interface RunOptions {
+  open?: boolean;
+  showLogs?: boolean;
+}
+
+export class ReportService {
+  private shouldOpen = true;
+
+  private shouldShowLogs = false;
+
+  private isReporterStarted = false;
+
+  private readonly reportsDir = path.join(this.rootDir, 'reports');
```

Comment:
it is really confusing for me to see that `this` keywords works correctly ouside of a function!

It is a TS magic :)